docker/docker.go

```diff
@@ -634,16 +638,53 @@ func getRegistryAuth(pr v3.PrivateRegistry) (string, error) {
 }
 
 func GetImageRegistryConfig(image string, prsMap map[string]v3.PrivateRegistry) (string, string, error) {
+	/*
+		Image can be passed as
+		- repo.com/foo/bar/rancher/rke-tools:v0.1.51
```

Comment:
Does adding a test make sense to test this function?

Reply:
@leodotcloud I can try adding a unittest. Don't think adding integration test would be possible with our current framework as we don't have long standing (and maintained) private repo